### PR TITLE
📖 docs update jobs.md

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Updates jobs.md with the new [test-infra](https://github.com/kubernetes/test-infra/pull/27421/files) jobs

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #6661

### CODE REVIEW

File: docs/book/src/reference/jobs.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Prow Presubmits:
   * GINKGO_FOCUS: `[IPv6] [PR-Informing]`, IP_FAMILY: `IPv6`
 * ✳️️ ✴️ [pull-cluster-api-e2e-full-main] `./scripts/ci-e2e.sh`
   * GINKGO_SKIP: `[PR-Blocking] [Conformance] [K8s-Upgrade] [IPv6]` (i.e. "no tags")
-* ✳️️ ✴️ [pull-cluster-api-e2e-workload-upgrade-1-24-latest-main] `./scripts/ci-e2e.sh` FROM: `stable-1.24` TO: `ci/latest-1.25`
+* ✳️️ ✴️ [pull-cluster-api-e2e-workload-upgrade-1-25-main] `./scripts/ci-e2e.sh` FROM: `stable-1.25` TO: `ci/latest-1.26`
```
Comment (sbueringer) [inline docs/book/src/reference/jobs.md:29]:
```suggestion
* ✳️️ ✴️ [pull-cluster-api-e2e-workload-upgrade-1-25-latest-main] `./scripts/ci-e2e.sh` FROM: `stable-1.25` TO: `ci/latest-1.26`
```
nit

Comment (sbueringer) [reply docs/book/src/reference/jobs.md:29]: Should also require a corresponding change in l.84

Comment (oscr) [reply docs/book/src/reference/jobs.md]: Thank you!

**Comment by sbueringer** [comment]
Thank you!!

/lgtm
/approve

**Pull request merged**